docs/packages/cluster/AUTOSCALING.md

```diff
@@ -0,0 +1,138 @@
+# Autoscaling
+
+Autoscaling occurs in the `cluster-autoscale` service, which runs every 15 seconds.
+
+## Why memory?
+
+The autoscaler uses CPU usage for GG nodes and memory usage for job nodes. This is because certain cloud providers like linode do not provide an actual value for the speed of the CPU, but rather the amount of cores. This is problematic because we use Nomad's API for determining the usage on any given node, and it returns its stats in MHz.
+
+## Hardware failover
+
+Before a job server provisioned, we don't know for sure what its specs will be because of the hardware failover system in `cluster-server-provision`. In the autoscaling process, all servers that aren't provisioned yet are assumed to have the specs of the first hardware option in the list.
```

Comment:
```suggestion
Before a job server is provisioned, we don't know for sure what its specs will be because of the hardware failover system in `cluster-server-provision`. In the autoscaling process, all servers that aren't provisioned yet are assumed to have the specs of the first hardware option in the list.
```